lib/sign.js

```diff
@@ -16,14 +17,36 @@ function sign (manifest, program, options, config) {
     AMOClient: DefaultAMOClient,
   }, config);
 
-  var client = new config.AMOClient({
-    apiKey: options.apiKey, apiSecret: options.apiSecret,
-    apiUrlPrefix: options.apiUrlPrefix,
-  });
+  return when.promise(function(resolve, reject) {
+
+    var missingOptions = [];
+    [
+      {value: options.apiKey, flag: '--api-key'},
+      {value: options.apiSecret, flag: '--api-secret'},
+    ]
+    .forEach(function(opt) {
```

Comment:
Yeah, I did hunt through the docs and source and I don't think commander handles it. Admittedly, required "options" is a bit odd but I don't think defining these as arguments is any nicer because you lose context like `key=...` and `secret=...`